lang-ref/part1.md

```diff
@@ -0,0 +1,941 @@
+```
+
+In the above code example, both **``obj1``** and **``obj2``** are objects of **``MyClass``**. Therefore *``obj1 === obj1``* returns true as both operands of the operation refer to the same memory location. Similarly *``obj1 === obj2``* returns false.
+ 
+The `‘==’` operator is used to check for contents of a structure type like Arrays.
+
+```
+//true
+boolean b3 = ([1,2,3] == [1,2,3]);
+
+//false
+boolean b4 = ([1,2,3] === [1,2,3]);
+```
+
+In the above code example, the first operation results in true as both the arrays have exactly the same content. However, the second operation is false because it is using the `‘===’` operator to check for memory location, and the two inline arrays point to different memory locations, even though their contents are the same.  
+
+In the case of floating point numbers, IEEE defines -0.0 and +0.0 as the same. Therefore applying `‘==’` to compare -0.0 and +0.0 would return a true, whereas *``-0.0 === +0.0``* would always be false as they are not identical.<br><br>
+
+## ``const`` and ``final``
+
+Ballerina supports ``const``, which is an immutable singleton type having a value known at compile time.
+
+```
+const MAX_VALUE = 1000;
+```
+
+Apart from ``const``, there is also support for ``final``. Variables defined with the ``final`` keyword cannot be reassigned after being initialized.
+
+```
+final string msg = loadMessage();
+```
+
+This is also applicable for class fields.<br><br>
+
+## Enumerations
+
+Ballerina supports the notion of Enumerations which is a shorthand way for expressing union of string constants.
+
+```
+enum Color {
+    RED, GREEN, BLUE
+}
+```
+
+Using the ``enum`` keyword, you can define this enumeration of three strings “RED”, “GREEN” and “BLUE”, which would be otherwise defined separately as a constant:
+
+```
+const RED = “RED”;
+const GREEN = “GREEN”;
+const BLUE = “BLUE”;
+
+type Color = RED|GREEN|BLUE;
+```
+ 
+You can also have an enumeration where members define associated string values explicitly.
+
+```
+enum Language {
+    ENG = "English",
+    TL = "Tamil",
+    SI = "Sinhala"
+}
+```
+<br><br>
+
+## ``match`` statement
+
+The ``match`` statement is like the switch case statement in C, but it is more flexible. You can use it to match values of different types.
+
+```
+const KEY = “xyzzy”;
+
+function mtest(any v) returns string {
+
+    match v {
+        17     =>  { return “number”; }
+        true   =>  { return “ boolean”; }
+        “str”  =>  { return “string” ; }
+        KEY    =>  { return “constant”; }
+        0|1    =>  { return “or”; }
+        _      =>  { return “any”; }
+    }
+}
+```
+
+The match statement defines multiple clause statement blocks with the `‘=>’` operator. The left hand side of the `‘=>’` operator in each clause is the pattern for value match. It can contain literals or identifiers referring to a constant.Additionally, it can also contain multiple patterns specified using `‘|’`, and `‘_’` to match any value that is not an error.  
```

Comment:
```suggestion
The match statement defines multiple clause statement blocks with the `‘=>’` operator. The left hand side of the `‘=>’` operator in each clause is the pattern for value match. It can contain literals or identifiers referring to a constant. Additionally, it can also contain multiple patterns specified using `‘|’`, and `‘_’` to match any value that is not an error.  
```